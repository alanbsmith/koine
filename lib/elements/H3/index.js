import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';

import { fontStyles, fontWeights } from '../../modifiers';
import gridScale from '../../utils/gridScale';

const modifierConfig = {
  ...fontStyles,
  ...fontWeights,
};

const H3 = styled.h3`
  font-size: ${props => gridScale(props, 4)};
  margin: 0;
  text-transform: capitalize;
  ${applyStyleModifiers(modifierConfig)}
`;

export default H3;
