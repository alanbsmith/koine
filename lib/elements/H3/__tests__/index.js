import React from 'react';
import H3 from '../index';
import renderer from 'react-test-renderer';


describe('A H3 Element', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <H3>Header Element</H3>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies lightWeight modifier', () => {
    const tree = renderer.create(
      <H3 modifiers={['lightWeight']}>Header Element</H3>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
