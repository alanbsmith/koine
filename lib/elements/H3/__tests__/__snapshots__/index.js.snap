// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A H3 Element applies lightWeight modifier 1`] = `
<h3
  className="sc-bdVaJa fiRqzR"
>
  Header Element
</h3>
`;

exports[`A H3 Element renders correctly 1`] = `
<h3
  className="sc-bdVaJa jBxvRE"
>
  Header Element
</h3>
`;
